lib/resourcebuilder/podspec.go

```diff
@@ -0,0 +1,47 @@
+package resourcebuilder
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// updatePodSpecWithProxy mutates the input podspec with proxy env vars for all init containers and containers
+// matching the container names.
+func updatePodSpecWithProxy(podSpec *corev1.PodSpec, containerNames []string, httpProxy, httpsProxy, noProxy string) error {
+	hasProxy := len(httpsProxy) > 0 || len(httpProxy) > 0 || len(noProxy) > 0
+	if !hasProxy {
+		return nil
+	}
+
+	for _, containerName := range containerNames {
+		found := false
+		for i := range podSpec.Containers {
+			if podSpec.Containers[i].Name != containerName {
+				continue
+			}
+			found = true
+
+			podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: httpProxy})
```

Comment:
Wouldn't this keep appending to the env list each sync cycle..?

Reply:
You're reading from raw bytes your read from a file, right?  How would that be cyclical?

Reply:
Oh, yeah missed it was right after read..